### DISCUSSION TITLE
Add documentation how to install extensions

### DISCUSSION THREAD

**Discussion by fishbone1** [ckan/ckan#7695] [Ideas]
Yes, I know, extensions should be documented by the provider. However I could not install a single extension this way.

They tell me I need a Python virtual env and also show how to activate it in some cases. Note that I'm not a Python dev.

The steps worked on an older server with source install. However on Ubuntu 22 with package install I have no idea. It seems that everything is owned by root. Should I

```
sudo . /usr/lib/ckan/default/bin/activate
```

? Or do I have to switch user to root like that:

```
sudo su - root
. /usr/lib/ckan/default/bin/activate
```

A little hint in the [package install docs](https://docs.ckan.org/en/2.10/maintaining/installing/index.html#package-install) would be great how to install extensions in general.

**Comment by kowh-ai** [comment]
Installing CKAN using the package format certainly is the quickest and easiest way to install CKAN as you have a fully functioning CKAN environment with the extensions DataStore and DataPusher running...however if you are wanting to do anything outside of that ie: installing more extensions or installing CKAN for development, then the recommendation is to install CKAN from source
